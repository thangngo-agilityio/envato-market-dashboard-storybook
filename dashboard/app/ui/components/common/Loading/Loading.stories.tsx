import { Meta, StoryObj } from '@storybook/react';
import Loading from '.';
import { Box } from '@chakra-ui/react';

const meta: Meta<typeof Loading> = {
  title: 'Custom Components/Loading',
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <Box>
        <Story />
      </Box>
    ),
  ],
  component: Loading,
};

export default meta;

type Story = StoryObj<typeof Loading>;

export const DefaultLoading: Story = {
  args: {},
};
