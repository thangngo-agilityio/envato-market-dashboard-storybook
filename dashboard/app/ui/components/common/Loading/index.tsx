'use client';

import { Flex, Spinner } from '@chakra-ui/react';

const Loading = () => (
  <Flex justifyContent="center">
    <Spinner />
  </Flex>
);

export default Loading;
