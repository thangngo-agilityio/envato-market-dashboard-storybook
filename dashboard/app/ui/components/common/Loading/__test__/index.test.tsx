import Loading from '..';

describe('Loading component', () => {
  it('Match with snapshot', () => {
    const { container } = render(<Loading />);

    expect(container).toMatchSnapshot();
  });
});
