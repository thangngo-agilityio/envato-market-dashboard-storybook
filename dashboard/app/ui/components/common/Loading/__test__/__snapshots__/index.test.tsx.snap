// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Match with snapshot 1`] = `
<div>
  <div
    class="css-1l4w6pd"
  >
    <div
      class="chakra-spinner css-nkpswe"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
